hypothesis-python/tests/cover/test_lookup_py38.py

```diff
@@ -113,3 +113,8 @@ class Node:
 @given(st.builds(Node))
 def test_can_resolve_recursive_dataclass(val):
     assert isinstance(val, Node)
+
+
+def test_can_register_new_type_for_typeddicts():
+    st.register_type_strategy(C, st.just("David Turner"))
```

Comment:
```python
from tests.common.utils import temp_registered

...

sentinel = object()
with temp_registered(C, st.just(sentinel)):
    assert st.from_type(C).example() is sentinel
```

Slightly more complicated, but it ensures that we don't "pollute" other tests if they're run in a different order, e.g. with [`pytest-randomly`](https://pypi.org/project/pytest-randomly/).